src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeEventInfo.cs

```diff
@@ -68,6 +69,12 @@ public override string ToString()
 
             return m_addMethod.GetParametersNoCopy()[0].ParameterType.FormatTypeName() + " " + Name;
         }
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
+            (RuntimeTypeMetadataUpdateHandler.HotReloadDeltaApplied && CacheEquals(obj));
+
+        public override int GetHashCode() => HashCode.Combine(m_token.GetHashCode(), RuntimeTypeHandle.GetModule(m_declaringType).GetHashCode());
```

Comment:
Is this going to be more expensive than it is today in the case where a delta hasn't been applied?

Reply:
I will run some perf tests

Reply:
The result looks bad, maybe should and call the old method if the Cache is not cleared 

|                   Method |        Job |         Toolchain |      Mean |    Error |   StdDev | Ratio | RatioSD |  Gen 0 | Allocated |
|------------------------- |----------- |------------------ |----------:|---------:|---------:|------:|--------:|-------:|----------:|
|       GetAMethodHashCode | Job-ARUJVB | \main\corerun.exe |  32.57 ns | 0.207 ns | 0.194 ns |  1.00 |    0.00 |      - |         - |
|       GetAMethodHashCode | Job-FDOPUD |   \pr\corerun.exe |  33.09 ns | 0.264 ns | 0.247 ns |  1.02 |    0.01 |      - |         - |
|                          |            |                   |           |          |          |       |         |        |           |
|    GetAllMethodsHashCode | Job-ARUJVB | \main\corerun.exe | 665.92 ns | 4.653 ns | 3.885 ns |  1.00 |    0.00 | 0.1259 |     528 B |
|    GetAllMethodsHashCode | Job-FDOPUD |   \pr\corerun.exe | 681.18 ns | 7.214 ns | 6.748 ns |  1.02 |    0.01 | 0.1259 |     528 B |
|                          |            |                   |           |          |          |       |         |        |           |
|     GetAPropertyHashCode | Job-ARUJVB | \main\corerun.exe |  32.31 ns | 0.226 ns | 0.201 ns |  1.00 |    0.00 |      - |         - |
|     GetAPropertyHashCode | Job-FDOPUD |   \pr\corerun.exe |  48.84 ns | 0.665 ns | 0.589 ns |  1.51 |    0.02 |      - |         - |
|                          |            |                   |           |          |          |       |         |        |           |
| GetAllPropertiesHashCode | Job-ARUJVB | \main\corerun.exe | 112.52 ns | 0.280 ns | 0.249 ns |  1.00 |    0.00 | 0.0191 |      80 B |
| GetAllPropertiesHashCode | Job-FDOPUD |   \pr\corerun.exe | 194.13 ns | 1.908 ns | 1.594 ns |  1.73 |    0.02 | 0.0191 |      80 B |
|                          |            |                   |           |          |          |       |         |        |           |
|        GetAFieldHashCode | Job-ARUJVB | \main\corerun.exe |  20.09 ns | 0.158 ns | 0.140 ns |  1.00 |    0.00 |      - |         - |
|        GetAFieldHashCode | Job-FDOPUD |   \pr\corerun.exe |  47.11 ns | 0.332 ns | 0.311 ns |  2.35 |    0.02 |      - |         - |
|                          |            |                   |           |          |          |       |         |        |           |
|     GetAllFieldsHashCode | Job-ARUJVB | \main\corerun.exe | 135.10 ns | 2.511 ns | 3.909 ns |  1.00 |    0.00 | 0.0439 |     184 B |
|     GetAllFieldsHashCode | Job-FDOPUD |   \pr\corerun.exe | 263.37 ns | 1.620 ns | 1.352 ns |  1.92 |    0.06 | 0.0439 |     184 B |

Reply:
After adding `MetadataUpdater` check when it is not supported the results looks comparable:

|                   Method |        Job |         Toolchain |      Mean |     Error |     StdDev | Ratio | RatioSD |  Gen 0 | Allocated |
|------------------------- |----------- |------------------ |----------:|----------:|-----------:|------:|--------:|-------:|----------:|
|     GetAPropertyHashCode | Job-CBGIOV | \main\corerun.exe |  34.69 ns |  0.526 ns |   0.492 ns |  1.00 |    0.00 |      - |         - |
|     GetAPropertyHashCode | Job-IRZIMW |   \pr\corerun.exe |  33.30 ns |  0.583 ns |   0.455 ns |  0.96 |    0.02 |      - |         - |
|                          |            |                   |           |           |            |       |         |        |           |
|        GetAFieldHashCode | Job-CBGIOV | \main\corerun.exe |  21.22 ns |  0.225 ns |   0.200 ns |  1.00 |    0.00 |      - |         - |
|        GetAFieldHashCode | Job-IRZIMW |   \pr\corerun.exe |  21.85 ns |  0.195 ns |   0.173 ns |  1.03 |    0.01 |      - |         - |

When `MetadataUpdater` is supported after updating the hashcode logic to `just combine the hashcode of the type and the token` as @jkotas suggested perf also improved 

|               Method |        Job |         Toolchain |     Mean |    Error |   StdDev | Ratio | RatioSD | Allocated |
|--------------------- |----------- |------------------ |---------:|---------:|---------:|------:|--------:|----------:|
| GetAPropertyHashCode | Job-QHEWUL | \main\corerun.exe | 32.41 ns | 0.273 ns | 0.255 ns |  1.00 |    0.00 |         - |
| GetAPropertyHashCode | Job-FJTZYE |   \pr\corerun.exe | 36.17 ns | 0.305 ns | 0.255 ns |  1.12 |    0.01 |         - |
|                      |            |                   |          |          |          |       |         |           |
|    GetAFieldHashCode | Job-QHEWUL | \main\corerun.exe | 20.04 ns | 0.283 ns | 0.236 ns |  1.00 |    0.00 |         - |
|    GetAFieldHashCode | Job-FJTZYE |   \pr\corerun.exe | 27.57 ns | 0.131 ns | 0.116 ns |  1.38 |    0.02 |         - |
|                      |            |                   |          |          |          |       |         |           |
|   GetAnEventHashCode | Job-QHEWUL | \main\corerun.exe | 20.31 ns | 0.136 ns | 0.120 ns |  1.00 |    0.00 |         - |
|   GetAnEventHashCode | Job-FJTZYE |   \pr\corerun.exe | 23.76 ns | 0.395 ns | 0.369 ns |  1.17 |    0.02 |         - |